.github/workflows/main.yml

```diff
@@ -108,3 +184,50 @@ jobs:
           cd tests/tlstest && \
           sed -n '/``` bash/,/```/{/```/d; p}' README.md > README.md.sh && \
           bash -euxo pipefail README.md.sh
+      - name: Check modification to Cargo.lock 
+        id: check-diff
+        run: |
+          file_changed=$(git diff --diff-filter=ACMUXTRD --name-only -- '**Cargo.lock' | tr '\n' ' ' | xargs)
+          echo "cargo-lock=$file_changed" >> $GITHUB_OUTPUT
+      - name: Output list of changed Cargo.lock
+        id: output-changed-cargo-lock
+        if: steps.check-diff.outputs.cargo-lock != ''
+        run: |
+            echo "::warning::Cargo.lock files modified" 
+            echo "::warning::Cargo.lock change list: ${{ steps.check-diff.outputs.cargo-lock }}" 
+      - name: Upload Cargo.lock files
+        id: upload-changed-cargo-lock
+        if: steps.check-diff.outputs.cargo-lock != ''
+        uses: actions/upload-artifact@v3
+        with: 
+          name: quickstart 
+          path: workspaces/**/Cargo.lock
+
+  cargo-lock-check:
+    needs: [linux, nitro, icecap, quickstart]
+    runs-on: ubuntu-latest
+    container:
```

Comment:
We probably don't need to be using our image for this workflow

Reply:
Removed all the container fields.